game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -2296,9 +2296,23 @@ public static void triggerUnitPlacement(final Set<TriggerAttachment> satisfiedTr
     }
   }
 
+  public static IntegerMap<Resource> findResourceIncome(final Set<TriggerAttachment> satisfiedTriggers,
+      final IDelegateBridge bridge) {
+    return triggerResourceChange(satisfiedTriggers, bridge, null, null, true, true, true, true, new StringBuilder());
+  }
+
   public static String triggerResourceChange(final Set<TriggerAttachment> satisfiedTriggers,
       final IDelegateBridge bridge, final String beforeOrAfter, final String stepName, final boolean useUses,
       final boolean testUses, final boolean testChance, final boolean testWhen) {
+    final StringBuilder strbuf = new StringBuilder();
+    triggerResourceChange(satisfiedTriggers, bridge, beforeOrAfter, stepName, useUses, testUses, testChance, testWhen,
+        strbuf);
+    return strbuf.toString();
+  }
+
+  private static IntegerMap<Resource> triggerResourceChange(final Set<TriggerAttachment> satisfiedTriggers,
+      final IDelegateBridge bridge, final String beforeOrAfter, final String stepName, final boolean useUses,
+      final boolean testUses, final boolean testChance, final boolean testWhen, final StringBuilder strbuf) {
```

Comment:
Since this method already has a lot of parameters, the new out parameter kind of gets buried.  Might want to consider returning a composite type that contains both the `IntegerMap` and the `String`.  Lombok should make creating such a value type pretty lightweight.

Otherwise, at least documenting that `strbuf` is an output parameter would be useful.  Also, a description in the Javadocs of what it will contain, or just renaming the variable, would be great, as `strbuf` isn't very descriptive.

Reply:
I'll just rename for now. I eventually want to refactor it further so the string for end turn report is done at the end.